From 3c36fbb71a018a4af17509bbe7ffc511f5099e2f Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 29 Nov 2023 13:53:42 +0100
Subject: [PATCH 429/954] feat(bl): align DATA section end on 2 bytes

NOT_UPSTREAMABLE

This is a workaround for Hyperflash boot on RevA and RevB revisions of
STM32MP25.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 1f04d409b6f67e920b0707982e0c109d454fd3a9 feat(bl): align DATA section end on 2 bytes]
Change-Id: I3242fb8fe593904616e0d8bb0b889906f625ce87
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/342677
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 include/common/bl_common.ld.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/common/bl_common.ld.h b/include/common/bl_common.ld.h
index b6dd0f0061..6c1c695015 100644
--- a/include/common/bl_common.ld.h
+++ b/include/common/bl_common.ld.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020-2022, ARM Limited and Contributors. All rights reserved.
+ * Copyright (c) 2020-2023, ARM Limited and Contributors. All rights reserved.
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -121,6 +121,7 @@
 	.data . : ALIGN(DATA_ALIGN) {			\
 		__DATA_START__ = .;			\
 		*(SORT_BY_ALIGNMENT(.data*))		\
+		. = ALIGN(2);				\
 		__DATA_END__ = .;			\
 	}
 
-- 
2.53.0

